### ISSUE TITLE
An empty message sent by e-mail as a comment to the ticket creates an error in the mail receiver, is not deleted and hangs in the Inbox, there is an error "If you reject the solution, you must specify a reason" in the log even if the ticket does not have a solution yet.

### ISSUE THREAD

**Issue by BorisNovg** [glpi-project/glpi#9726]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

9.5.6

### Bug description

The user to whom we provide support services responds to a comment on the ticket by e-mail, however, the comment does not appear in the web interface of our GLPI-based support system, and the letter remains indefinitely in the Inbox of the mail receiver.

The analysis of the letter shows that the user just accidentally deleted the line ``=-=-=-= To answer by email, write above this line =-=-=-=``, habitually removing unnecessary lines ``From:``
``Sent:`` ``To:`` ``Subject:`` from the header of the letter.

The problem is not reproduced if the user writes the FIRST comment to the ticket (in this case, a letter with a comment without the line ``=-=-=-= To answer by email, write above this line =-=-=-=`` is processed in a completely normal way and appears in the web interface of our GLPI-based support system).

But if there are already at least 3 comments in the dialog with the user, an error occurs.

In addition, a very strange error is recorded in the log, clearly indicating an incorrectly compiled algorithm for processing mail messages:  "If you reject the solution, you must specify a reason". Please note that the ticket is not closed at this moment, and the error message indicates that the algorithm got into the part of the code that gives an error with this text, obviously by accident.
![glpi-comment-1](https://user-images.githubusercontent.com/39726474/137906434-bcd1d173-36cd-4b4d-a0ec-8a99744c0665.gif)

### Relevant log output

```shell
cron.log
``
2021-10-19 12:31:01 [@help.admreg.nov.ru]
External #1: Run mailgate
2021-10-19 12:31:02 [@help.admreg.nov.ru]
If you reject the solution, you must specify a reason

``

php-errors.log
``
(At this point in time, nothing is written to the log)
``
Home-Setup-Automatic actions-mailgate-Logs-12:31
``
0 Run mode: CLI
0 Collect mails from XXX@XXX.ru ({10.10.1.72/imap/ssl/novalidate-cert/norsh})
1 Number of messages: available=1, already imported=0, retrieved=1, refused=0, errors=1, blacklisted=0
1 Action completed, fully processed
``
```

### Steps To reproduce

1. End User action: Write a support letter.
2. Action of the support system specialist: add a comment.
3. End user action: Reply to a comment by mail.
4. Action of the support system specialist: add a comment.
5. End user action: Reply to a comment by mail, delete line ``=-=-=-= To answer by email, write above this line =-=-=-=`` when writing a response.

### Your GLPI setup information

```markdown
GLPI 9.5.6 ( => /usr/share/glpi)
Installation mode: RPM

Server
 
Operating system: Linux help.admreg.nov.ru 3.10.0-1160.42.2.el7.x86_64 #1 SMP Tue Sep 7 14:49:57 UTC 2021 x86_64
PHP 8.0.11 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, bcmath, bz2,
	calendar, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, intl, json, ldap, libxml,
	mbstring, mysqli, mysqlnd, openssl, pcre, pdo_mysql, pdo_sqlite, posix, session, shmop, soap, sockets, sodium, sqlite3,
	standard, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib)
Setup: max_execution_time="600" memory_limit="512M" post_max_size="10500000" safe_mode="" session.save_handler="files"
	upload_max_filesize="10500000" 
Software: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/8.0.11 ()
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36
Server Software: MariaDB Server
	Server Version: 10.5.12-MariaDB
	Server SQL Mode: 
	Parameters: saglpi@127.0.0.1/glpi
	Host info: 127.0.0.1 via TCP/IP
	
PHP version is at least 7.2.0 - Perfect!PHP version is at least 7.2.0 - Perfect!
Sessions support is available - Perfect!Sessions support is available - Perfect!
Allocated memory > 64 Mio - Perfect!Allocated memory > 64 Mio - Perfect!
mysqli extension is installedmysqli extension is installed
ctype extension is installedctype extension is installed
fileinfo extension is installedfileinfo extension is installed
json extension is installedjson extension is installed
mbstring extension is installedmbstring extension is installed
iconv extension is installediconv extension is installed
zlib extension is installedzlib extension is installed
curl extension is installedcurl extension is installed
gd extension is installedgd extension is installed
simplexml extension is installedsimplexml extension is installed
intl extension is installedintl extension is installed
ldap extension is installedldap extension is installed
apcu extension is installedapcu extension is installed
Zend OPcache extension is installedZend OPcache extension is installed
xmlrpc extension is installedxmlrpc extension is installed
exif extension is installedexif extension is installed
zip extension is installedzip extension is installed
bz2 extension is installedbz2 extension is installed
sodium extension is installedsodium extension is installed
Database version seems correct (10.5.12) - Perfect!Database version seems correct (10.5.12) - Perfect!
Timezones seems loaded in databaseTimezones seems loaded in database
The log file has been created successfully.The log file has been created successfully.
Write access to /var/lib/glpi/files/_cache has been validated.Write access to /var/lib/glpi/files/_cache has been validated.
Write access to /etc/glpi has been validated.Write access to /etc/glpi has been validated.
Write access to /var/lib/glpi/files/_cron has been validated.Write access to /var/lib/glpi/files/_cron has been validated.
Write access to /var/lib/glpi/files has been validated.Write access to /var/lib/glpi/files has been validated.
Write access to /var/lib/glpi/files/_dumps has been validated.Write access to /var/lib/glpi/files/_dumps has been validated.
Write access to /var/lib/glpi/files/_graphs has been validated.Write access to /var/lib/glpi/files/_graphs has been validated.
Write access to /var/lib/glpi/files/_lock has been validated.Write access to /var/lib/glpi/files/_lock has been validated.
Write access to /var/lib/glpi/files/_pictures has been validated.Write access to /var/lib/glpi/files/_pictures has been validated.
Write access to /var/lib/glpi/files/_plugins has been validated.Write access to /var/lib/glpi/files/_plugins has been validated.
Write access to /var/lib/glpi/files/_rss has been validated.Write access to /var/lib/glpi/files/_rss has been validated.
Write access to /var/lib/glpi/files/_sessions has been validated.Write access to /var/lib/glpi/files/_sessions has been validated.
Write access to /var/lib/glpi/files/_tmp has been validated.Write access to /var/lib/glpi/files/_tmp has been validated.
Write access to /var/lib/glpi/files/_uploads has been validated.Write access to /var/lib/glpi/files/_uploads has been validated.
Write access to /var/lib/glpi/files/_marketplace has been validated.Write access to /var/lib/glpi/files/_marketplace has been validated.
SELinux configuration is OK.SELinux configuration is OK.

GLPI constants
 
GLPI_ROOT: /usr/share/glpi
GLPI_CONFIG_DIR: /etc/glpi
GLPI_HTMLAWED: /usr/share/glpi/vendor/htmlawed/htmlawed/htmLawed.php
GLPI_MARKETPLACE_DIR: /var/lib/glpi/files/_marketplace
GLPI_VAR_DIR: /var/lib/glpi/files
GLPI_LOG_DIR: /var/log/glpi
GLPI_FONT_FREESANS: /usr/share/fonts/gnu-free/FreeSans.ttf
GLPI_SYSTEM_CRON: 1
GLPI_INSTALL_MODE: RPM
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_USE_IDOR_CHECK: 1
GLPI_IDOR_EXPIRES: 7200
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: 
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_PRERELEASES: 
GLPI_MARKETPLACE_ALLOW_OVERRIDE: 1
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: 1
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_FORCE_EMPTY_SQL_MODE: 1
GLPI_DOC_DIR: /var/lib/glpi/files
GLPI_CACHE_DIR: /var/lib/glpi/files/_cache
GLPI_CRON_DIR: /var/lib/glpi/files/_cron
GLPI_DUMP_DIR: /var/lib/glpi/files/_dumps
GLPI_GRAPH_DIR: /var/lib/glpi/files/_graphs
GLPI_LOCAL_I18N_DIR: /var/lib/glpi/files/_locales
GLPI_LOCK_DIR: /var/lib/glpi/files/_lock
GLPI_PICTURE_DIR: /var/lib/glpi/files/_pictures
GLPI_PLUGIN_DOC_DIR: /var/lib/glpi/files/_plugins
GLPI_RSS_DIR: /var/lib/glpi/files/_rss
GLPI_SESSION_DIR: /var/lib/glpi/files/_sessions
GLPI_TMP_DIR: /var/lib/glpi/files/_tmp
GLPI_UPLOAD_DIR: /var/lib/glpi/files/_uploads
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: /usr/share/glpi/locales
GLPI_VERSION: 9.5.6
GLPI_SCHEMA_VERSION: 9.5.6
GLPI_MIN_PHP: 7.2.0
GLPI_YEAR: 2021

Libraries
 
htmlawed/htmlawed version 1.2.5 in (/usr/share/glpi/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.1.6 in (/usr/share/glpi/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.6 in (/usr/share/glpi/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.3.5 in (/usr/share/glpi/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/usr/share/glpi/vendor/michelf/php-markdown/Michelf)
true/punycode in (/usr/share/glpi/vendor/true/punycode/src)
iamcal/lib_autolink in (/usr/share/glpi/vendor/iamcal/lib_autolink)
sabre/dav in (/usr/share/glpi/vendor/sabre/dav/lib/DAV)
sabre/http in (/usr/share/glpi/vendor/sabre/http/lib)
sabre/uri in (/usr/share/glpi/vendor/sabre/uri/lib)
sabre/vobject in (/usr/share/glpi/vendor/sabre/vobject/lib)
laminas/laminas-cache in (/usr/share/glpi/vendor/laminas/laminas-cache/src)
laminas/laminas-i18n in (/usr/share/glpi/vendor/laminas/laminas-i18n/src)
laminas/laminas-serializer in (/usr/share/glpi/vendor/laminas/laminas-serializer/src)
monolog/monolog in (/usr/share/glpi/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/usr/share/glpi/vendor/sebastian/diff/src)
elvanto/litemoji in (/usr/share/glpi/vendor/elvanto/litemoji/src)
symfony/console in (/usr/share/glpi/vendor/symfony/console)
scssphp/scssphp in (/usr/share/glpi/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/usr/share/glpi/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/usr/share/glpi/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/usr/share/glpi/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/usr/share/glpi/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/usr/share/glpi/vendor/ramsey/uuid/src)
psr/log in (/usr/share/glpi/vendor/psr/log/Psr/Log)
psr/simple-cache in (/usr/share/glpi/vendor/psr/simple-cache/src)
mexitek/phpcolors in (/usr/share/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/usr/share/glpi/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/usr/share/glpi/vendor/guzzlehttp/psr7/src)
wapmorgan/unified-archive in (/usr/share/glpi/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/usr/share/glpi/vendor/paragonie/sodium_compat/src)
phpCas version 1.4.0 in (/usr/share/pear)

LDAP directories

SQL replicas
 
Not active

Notifications
 
Way of sending emails: SMTP+TLS (anonymous@10.10.1.72)

Mails receivers
 
Name: 'XXX@XXX.ru' Active: Yes
	Server: '{10.10.1.72/imap/ssl/novalidate-cert/norsh}' Login: 'XXX@XXX.ru' Password: Yes

Plugins list
 
	news                 Name: Alerts                         Version: 1.9.0      State: Enabled
	behaviors            Name: Behaviours                     Version: 2.5.0      State: Enabled
	dashboard            Name: Dashboard                      Version: 1.0.2      State: Enabled
	escalade             Name: Escalation                     Version: 2.6.2      State: Enabled
	formcreator          Name: Form Creator                   Version: 2.12.2     State: Enabled
	mreporting           Name: More Reporting                 Version: 1.7.3      State: Enabled
	moreticket           Name: More ticket                    Version: 1.6.1      State: Enabled
	mydashboard          Name: My Dashboard                   Version: 1.8.2      State: Enabled
	mycustomview         Name: MyCustomView                   Version: 1.2.1      State: Enabled
	ocsinventoryng       Name: OCS Inventory NG               Version: 1.7.3      State: Enabled
	reports              Name: Reports                        Version: 1.14.1     State: Enabled
	screenshot           Name: Screenshot                     Version: 1.1.4      State: Enabled

OCS Inventory NG
Host: 10.10.0.72
Connection: Ok
Use the OCSNG software dictionary: Yes
```

**Comment by cedric-anne** [comment]
> Accordingly, the problem narrows down to the reaction to an empty message - such a message is not deleted and hangs in the Inbox!

You are right, it may cause issues. When a ticket or a followup `add` operation returns a falsy result, which happens in this case, it is considered as an error (`NotImportedEmail::FAILED_OPERATION`), but email is not moved into the `refused` folder. It is not so easy, right now to know why the `add` operation failed.

In the collector cofiguration, you can use the `Collect only unread mail` configuration. It should prevent this kind of message to be fetched each time.

**Comment by k0xak** [comment]
I dont get it.

We have fully working 9.4 where email followup works.
When we upgrade to 9.5 emial followups stops to work and emails are stuck in inbox

Created emails are same, we reply in same way - it just down work.
Even "=-=-=-= To answer by email, write above this line =-=-=-=" exists in the reply.


Mailgate response:
If you reject the solution, you must specify a reason

**Comment by BorisNovg** [comment]
> Created emails are same, we reply in same way - it just down work. Even "=-=-=-= To answer by email, write above this line =-=-=-=" exists in the reply.
> Mailgate response: If you reject the solution, you must specify a reason

k0xak, I think the reason for the "If you reject the solution, you must specify a reason" response is that the message handler concludes that a BLANK letter has arrived at glpi. And this conclusion is made due to the fact that the letter does not contain the line "=_=_=_= To answer by email, write under this line =_=_=_=", meaning the end of the quote. The response "If you reject the solution, you must specify a reason" is output precisely as a reaction to an EMPTY incoming message.

Tip - Look for the reason why "=-=-=-= To answer by email, write above this line =-=-=-=" is present in messages and "=_=_=_= To answer by email, write under this line =_=_=_=" is missing. Most likely, you should pay the attention of users so that when responding to a message from glpi, they do not delete the "unnecessary" message tail along with the line "=_=_=_= To answer by email, write under this line =_=_=_=". Alternatively, they can remove the "unnecessary" tail, but strictly ensure that the string "=_=_=_= To answer by email, write under this line =_=_=_=" remains in the response (not deleted by the user).

**Comment by cconard96** [comment]
Related to #10792
Proposed fix for GLPI 10 (done for next major version because it is a change in expected behavior): #10797

**Comment by cconard96** [comment]
Fix #10797 was merged and will be included with 10.1.0.
